ql/src/java/org/apache/hadoop/hive/ql/exec/tez/TezTask.java

```diff
@@ -384,24 +382,28 @@ private void collectCommitInformation(TezWork work) throws IOException, TezExcep
               // get all target tables this vertex wrote to
               List<String> tables = new ArrayList<>();
               for (Map.Entry<String, String> entry : jobConf) {
```

Comment:
Do we have a faster solution for this? The `jobConf` could be very-very big

Reply:
Good question. Don't think there's a faster way if we want to keep this generic and work for all existing and future iceberg properties. I did find a convenience method though which can do this for us without a loop:
`public Map<String, String> getPropsWithPrefix(String confPrefix)`, but it also loops through the keys internally.

If we are concerned about performance, we can find out which exact properties are needed to be propagated, hardcode those and simply pass only those few without the looping.

Reply:
Merged the two loops.